Enhancements/remove rx dart

dw2kim opened the pull request.

dw2kim committed:
Fix ios runner issue

dw2kim committed:
Remove rxdart

dw2kim committed:
Removed bloc and add callback function for formData

dw2kim committed:
Updated ReadMe

dw2kim committed:
Clean up example code

MisterJimson commented on lib/src/json_schema_form.dart:
```diff
@@ -1,30 +1,61 @@
-  final JsonSchemaBloc jsonSchemaBloc;
+  final String jsonSchema;
+  final String uiSchema;
+  final Function(String) onFormDataChanged;
```
`final void Function(String)` would make the return type of this function explicit instead of dynamic as it is right now.

MisterJimson commented on lib/src/json_schema_form.dart:
```diff
@@ -1,30 +1,61 @@
+  JsonSchemaForm(
+      {@required this.jsonSchema,
+      @required this.uiSchema,
+      this.onFormDataChanged});
```
onFormDataChanged is not required but the other 2 params are. Is there a reason for this?

dw2kim replied:
I was thinking that the user can get the formData from the property whenever they want instead of depending on the callback. But I see that the "formData" is not there in JsonSchemaForm but in its state. I am down to change it to be required (not had strong feeling towards not to have it required).

MisterJimson replied:
Lets make it required for now, I like the idea of being able to get the current data whenever as a property. I'll create an issue for that.

MisterJimson commented on lib/src/json_schema_form.dart:
```diff
@@ -1,30 +1,61 @@
-  final Schema schema;
+  final String jsonSchema;
+  final String uiSchema;
+  final Function(String) onFormDataChanged;
```
Same comment about function return type

MisterJimson commented on lib/src/json_schema_form.dart:
```diff
@@ -1,30 +1,61 @@
+  _JsonSchemaFormState({
+    @required this.jsonSchema,
+    @required this.uiSchema,
+    this.onFormDataChanged,
```
Same comment about @required

MisterJimson commented on lib/src/json_schema_form.dart:
```diff
@@ -79,9 +110,10 @@ class _JsonSchemaFormState extends State<JsonSchemaForm> {
+
+                        setState(() {
+                          onFormDataChanged(formData);
+                        });
```
2 comments here.

Is setState needed? I don't see any of _JsonSchemaFormState's state that would be mutated in this block.

Does `onFormDataChanged` communicate the use of the callback the best? What do you think of `onSubmit` instead?

dw2kim replied:
1. No. This is mistaken, I will remove the `setState`.
2. I think both works fine but feel `onFormDataChanged` is a little more specific what it is expected.

MisterJimson reviewed (changes requested):
Looking pretty solid. Some feedback noted.

dw2kim committed:
PR Feedbacks 

1. Add Required for onFormDataChanged callback
2. Remove setState for FormDataChanged
3. Make the return type of this function explicit for onFormDataChanged
